lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -302,6 +302,108 @@ class ConvertValsemVariantAtenIndexPutImplOp
+                  ConversionPatternRewriter &rewriter) const override {
+    // Verify types
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    // For debugging
+
+    op->getParentOfType<ModuleOp>().dump();
+
+    Value grad_output = adaptor.grad_output();
+    Value input = adaptor.self();
+    MLIRContext *context = op->getContext();
+    RankedTensorType inputType = input.getType().cast<RankedTensorType>();
+    RankedTensorType gradType = grad_output.getType().cast<RankedTensorType>();
+    SmallVector<Value> inputShape = getTensorSizes(rewriter, loc, input);
+    Type inputEType = inputType.getElementType();
+
+    // Get values
+    SmallVector<Value> kernelSize;
+    if (!getListConstructElements(op.kernel_size(), kernelSize))
+      return rewriter.notifyMatchFailure(op, "kernel size not constructed from ListConstruct");
+    SmallVector<Value> stride;
+    if (!getListConstructElements(op.stride(), stride))
+      return rewriter.notifyMatchFailure(op, "stride not constructed from ListConstruct");
+    SmallVector<Value> padding;
+    if (!getListConstructElements(op.padding(), padding))
+      return rewriter.notifyMatchFailure(op, "padding not constructed from ListConstruct");
+    SmallVector<Value> dilation;
+    if (!getListConstructElements(op.dilation(), dilation))
+      return rewriter.notifyMatchFailure(op, "dilation not constructed from ListConstruct");
+
+    // The element type of the `self` and `values` should be same.
+    if (inputType.getElementType() != gradType.getElementType())
+      return rewriter.notifyMatchFailure(
+          op, "Input element type should be same as the grad_output element type.");
+
+    // Add asserts/compile-time checks for values
+    // TODO
+
+    // Create empty output tensor
+    // Value output = rewriter.create<linalg::InitTensorOp>(loc, selfShape, selfEType);
+
+    // Add in indices
+    Attribute attrs[2];
+    if (inputEType.isa<mlir::FloatType>()) {
+      attrs[0] = rewriter.getFloatAttr(inputEType, 0);
+      attrs[1] = rewriter.getFloatAttr(inputEType, 1);
+    } else if (inputEType.isa<mlir::IntegerType>()) {
+      attrs[0] = rewriter.getIntegerAttr(inputEType, 0);
+      attrs[1] = rewriter.getIntegerAttr(inputEType, 1);
+    } else {
+      return rewriter.notifyMatchFailure(op, "unsupported dtype");
+    }
+
+    Value output = createZeroInitTensor(rewriter, loc, inputShape, inputEType);
+    //Value numTensorElements = rewriter.create<AtenNumelOp>(loc, input);
+    //rewriter.create<AtenZerosOp>(loc, inputType, numTensorElements, inputEType);
+
+    //Value indicesList;
+    //if (!getListConstructElements(adaptor.indices(), indicesList))
+     // return rewriter.notifyMatchFailure(op, "indices not constructed from ListConstruct");
+    
+
+    ValueTensorType indexType =
+        adaptor.indices().getType().cast<ValueTensorType>();
+
+    Value expandedIndexSizes{indexType.getSizes(), 1};
```

Comment:
We initialize the vector this way. Change the `Value expanded...` -> `SmallVector<Value> expanded...`

Reply:
I updated those as we discussed.